dorado/particle_track.py

```diff
@@ -403,14 +373,160 @@ def __init__(self, params):
+        :obj:`dorado.particle_track.run_iteration`.
+
+        Walk data is stored within :obj:`dorado.particle_track.Particles` as
+        `Particles.walk_data` so if this method is called in succession, even
+        without the `previous_walk_data` flag, the new particle seed locations
+        will be appended to the `walk_data` attribute of
+        :obj:`dorado.particle_track.Particles`.
+
+        **Inputs** :
+
+            Np_tracer : `int`
+                Number of particles to generate.
+
+            seed_xloc : `list`
+                List of x-coordinates over which to initially distribute the
+                particles.
+
+            seed_yloc : `list`
+                List of y-coordinates over which to initially distribute the
+                particles.
+
+            method : `str`, optional
+                Specify the type of particle generation you wish to use.
+                Current options are 'random' and 'exact' for seeding particles
+                either randomly across a set of points, or exactly at a set
+                of defined locations. If unspecified, the default is 'random'.
+
+            previous_walk_data : `dict`, optional
+                Dictionary of all prior x locations, y locations, and travel
+                times. This input parameter should only be used if 'new' or
+                'indpendent' walk data exists (e.g. data loaded from a .json
+                file). Otherwise the walk data stored in `self.walk_data` is
+                likely to contain the previous information. Order of indices is
+                previous_walk_data[field][particle][iter], where e.g.
+                ['travel_times'][5][10] is the travel time of the 5th particle
+                at the 10th iteration.
+
+        """
+        # if the values in self are invalid the input checked will catch them
+        # do input type checking
+        Np_tracer, seed_xloc, seed_yloc = gen_input_check(Np_tracer,
+                                                          seed_xloc,
+                                                          seed_yloc,
+                                                          method)
+
+        init_walk_data = dict()  # create init_walk_data dictionary
+
+        # initialize new travel times list
+        new_start_times = [0.]*Np_tracer
+
+        if method == 'random':
+            # create random start indices based on x, y locations and np_tracer
+            new_start_xindices = [lw.random_pick_seed(seed_xloc) for x
+                                  in list(range(Np_tracer))]
+            new_start_yindices = [lw.random_pick_seed(seed_yloc) for x
+                                  in list(range(Np_tracer))]
+
+        elif method == 'exact':
+            # step through seed lists to create exact list of start locations
+            # if number of particles exceeds length of seed lists, loop back
+            # to beginning of the seed lists and continue from there
```

Comment:
Didn't consider what would happen if `Np_tracer != len(seed_xloc)` in the exact scenario, good catch. This seems like a good solution.